cwltool/process.py

```diff
@@ -804,6 +810,98 @@ def _init_job(
+                                    high = cast(
+                                        Union[int, float],
+                                        constraint.get("high", math.inf),
+                                    )
+                                    matched = value >= low and value <= high
+                                elif constraint["class"] == "realInterval":
+                                    if not isinstance(value, (int, float)):
+                                        raise SourceLine(
+                                            constraint,
+                                            None,
+                                            WorkflowException,
+                                            runtime_context.debug,
+                                        ).makeError(
+                                            "realInterval parameter restriction is only valid for inputs of type 'int', 'float', and 'double'; "
+                                            f"instead got {type(value)}: {value}."
+                                        )
+                                    low = cast(
+                                        Union[int, float],
+                                        constraint.get("low", -math.inf),
+                                    )
+                                    high = cast(
+                                        Union[int, float],
+                                        constraint.get("high", math.inf),
+                                    )
+                                    low_inclusive = constraint.get(
+                                        "low_inclusive", True
+                                    )
+                                    high_inclusive = constraint.get(
+                                        "high_inclusive", True
+                                    )
+                                    check_low = (
+                                        value >= low if low_inclusive else value > low
+                                    )
+                                    check_high = (
+                                        value <= high if low_inclusive else value < high
+                                    )
+                                    matched = check_low and check_high
+                                elif constraint["class"] == "regex":
+                                    rpattern = constraint["rpattern"]
+                                    quoted_value = json.dumps(value)
+                                    matched = cast(
+                                        bool,
+                                        execjs(
+                                            f"/{rpattern}/.test({quoted_value})",
+                                            "",
+                                            runtime_context.eval_timeout,
+                                            runtime_context.force_docker_pull,
+                                        ),
```

Comment:
Couldn't we use `re.search(rpattern, quoted_value)` or another Python statement here?

Reply:
We could, but then we'd have to codify the Python RegEx rules into the CWL standard; it is simpler to use the ECMAScript 5.1 regex functionality as that matches what we already use for CWL Expressions

Reply:
Good point, thanks for explaining it :+1: